### PR TITLE
New module using what-vpn

### PR DESCRIPTION
**Pull request by lukigruszka**
Makes use of [what-vpn](https://github.com/dlenski/what-vpn/) project, enabling Artemis to identify SSL VPN servers.
Takes IPs as input, subject to change (other alternatives are domains or HTTP servises)

### CODE REVIEW

File: extra_modules_config.py [hunk 1]
```diff
@@ -54,5 +54,13 @@ class ExtraModulesConfig:
         ),
     )
 
+    # Timeout counted in seconds, after which the what-vpn module terminates a connection and starts using the next sniffer.
+    # Some of VPN gateways do not respond in any way to the HTTP(S) requests, so the timeout variable should be optimized in 
+    # order to avoid false negatives while not blocking the task for too long.
+    WHATVPN_TIMEOUT_S = decouple.config(
```
Comment (kazet) [inline extra_modules_config.py:60]: nitpick: _S -> _SECOND

File: karton_whatvpn/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+FROM certpl/artemis:latest
+
+RUN apk add git
+RUN pip install requests
+RUN pip3 install https://github.com/dlenski/what-vpn/archive/master.zip
```
Comment (kazet) [inline karton_whatvpn/Dockerfile:5]: pin versions

File: karton_whatvpn/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,9 @@
+FROM certpl/artemis:latest
+
+RUN apk add git
+RUN pip install requests
```
Comment (kazet) [inline karton_whatvpn/Dockerfile:4]: requests is already in the artemis image

File: karton_whatvpn/karton_whatvpn.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+@load_risk_class.load_risk_class(load_risk_class.LoadRiskClass.LOW)
+class WhatVPN(ArtemisBase):
+    """ 
+    Runs what-vpn -> SSL VPN identifier
+    """
+    
+    identity = "what-vpn"
+    filters = [
+        {"type": TaskType.IP.value}
```
Comment (kazet) [inline karton_whatvpn/karton_whatvpn.py:22]: and why not domain? or even SERVICE?

File: karton_whatvpn/karton_whatvpn.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+                host
+            ],
+            capture_output=True
+        )
+        output = output.stdout.decode("utf-8")
+        detected_vpn = []
+
+        error_messages = ["error", "timeout", "no match"]
+        if any(msg in output for msg in error_messages):
+            status = TaskStatus.OK
```
Comment (kazet) [inline karton_whatvpn/karton_whatvpn.py:41]: TaskStatus.ERROR

**Comment by kazet** [comment]
fix linter ;)

File: extra_modules_config.py (hunk 1 above)
Comment (lukigruszka) [reply extra_modules_config.py]: Changed all occurrences

File: karton_whatvpn/Dockerfile (hunk 2 above)
Comment (lukigruszka) [reply karton_whatvpn/Dockerfile]: pinned latest hash on master branch

File: karton_whatvpn/Dockerfile (hunk 3 above)
Comment (lukigruszka) [reply karton_whatvpn/Dockerfile]: removed

File: karton_whatvpn/karton_whatvpn.py (hunk 5 above)
Comment (lukigruszka) [reply karton_whatvpn/karton_whatvpn.py:36]: changed the logic to return separate OK and ERROR statuses

File: karton_whatvpn/karton_whatvpn.py (hunk 4 above)
Comment (lukigruszka) [reply karton_whatvpn/karton_whatvpn.py]: I'll stay with IP task type as profit from scanning all discovered ports is debatable while increase number of sent requests  is noticeable.
Might change in the future based on observations.

**Review by kazet — APPROVED** [review]

**Pull request merged**